docs/contracts/hash-consensus.md

```diff
@@ -16,3 +16,428 @@ the slot returned from `IReportAsyncProcessor.getLastProcessingRefSlot`.
+```solidity
+function getChainConfig() external view returns (
+    uint256 slotsPerEpoch,
+    uint256 secondsPerSlot,
+    uint256 genesisTime
+)
+```
+
+#### Returns
+
+| Name             | Type      | Description                                                |
+| ---------------- | --------- | ---------------------------------------------------------- |
+| `slotsPerEpoch`  | `uint256` | Number of slots per epoch, `32` by default                 |
+| `secondsPerSlot` | `uint256` | The time allocated for each slot, `12` by default          |
+| `genesisTime`    | `uint256` | Genesis time, `1606824023` by default, more [here](https://blog.ethereum.org/2020/11/27/eth2-quick-update-no-21)          |
+
+### getFrameConfig()
+
+Returns the time-related configuration.
+
+```solidity
+function getFrameConfig() external view returns (
+    uint256 initialEpoch,
+    uint256 epochsPerFrame,
+    uint256 fastLaneLengthSlots
+)
+```
+
+#### Returns
+
+| Name                  | Type      | Description                                                           |
+| --------------------- | --------- | --------------------------------------------------------------------- |
+| `initialEpoch`        | `uint256` | Epoch of the frame with zero index                                    |
+| `epochsPerFrame`      | `uint256` | Length of a frame in epochs.                                          |
+| `fastLaneLengthSlots` | `uint256` | Length of the fast lane interval in slots; see `getIsFastLaneMember`. |
+
+### getCurrentFrame()
+
+Returns the current reporting frame.
+
+```solidity
+function getCurrentFrame() external view returns (
+    uint256 refSlot,
+    uint256 reportProcessingDeadlineSlot
+)
+```
+
+#### Returns
+
+| Name                           | Type      | Description                                                                          |
+| ------------------------------ | --------- | ------------------------------------------------------------------------------------ |
+| `refSlot`                      | `uint256` | The frame's reference slot: if the data the consensus is being reached upon includes or depends on any onchain state, this state should be queried at the reference slot. If the slot contains a block, the state should include all changes from that block. |
+| `reportProcessingDeadlineSlot` | `uint256` | The last slot at which the report can be processed by the report processor contract. |
+
+### getInitialRefSlot()
+
+Returns the earliest possible reference slot, i.e. the reference slot of the reporting frame with zero index.
+
+```solidity
+function getInitialRefSlot() external view returns (uint256)
+```
+
+### getIsMember()
+Returns whether the given address is currently a member of the consensus.
+```solidity
+function getIsMember(address addr) external view returns (bool)
+```
+
+### getIsFastLaneMember()
+
+Returns whether the given address is a fast lane member for the current reporting frame.
```

Comment:
it's a good piece of explanation, however, should be move upper I bet in the specific chapter